src/darwin/Framework/CHIPTests/MTROTAProviderTests.m

```diff
@@ -888,7 +988,472 @@ - (void)test005_DoBDXTransferAllowUpdateRequest
     // to _any_ of the above expectations.
     [self waitForExpectations:@[ announceResponseExpectation ] timeout:kTimeoutInSeconds];
 }
-#endif // ENABLE_TEST_005
+
+- (void)test007_DoBDXTransferWithTwoOTARequesters
+{
+    // In this test we do the following:
+    //
+    // 1) Check if the ota image files and raw image files required for this test exist for both OTA requestors
+    // 2) Advertise ourselves to the first OTA requestor.
+    // 3) When device1 queries for an image, pass the image path for the ota file generated beforehand as a pre-requisite for first
+    // OTA requestor 4) When device1 tries to start a bdx transfer, respond with success for the first OTA requestor
+    //    Set queryHandler to return busy with a busy delay of 30 secs and advertise ourselves to the second OTA requestor
+    // 5) When second OTA requestor queries send image available with busy delay of 30 secs indicating the provider is busy.
+    // 6) Send the data as the BDX transfer proceeds for the first OTA requestor
+    // 7) When BDX transfer ends, set up the query handler to return available for second OTA requestor and populate all the other
+    // BDX handlers
+    //    to handle BDX for the second requestor.
+    // 8) When device2 queries for an image, pass the image path for the ota file generated beforehand as a pre-requisite for second
```

Comment:
Again, something went very wrong with the wrapping starting at this item.

Reply:
cleaned this up as much as i could. sorry for the messed up formatting